### PR TITLE
Mapnik omnivore refactor

### PR DESCRIPTION
**Pull request by GretaCB**
Prep Studio for integrating the new mapnik-omnivore refactor

### CODE REVIEW

**Comment by springmeyer** [comment]
will this address #1156?

**Comment by GretaCB** [comment]
@springmeyer Yes. Studio no longer directly calls any mapnik-omnivore functions, so this error shouldn't happen anymore. Also, added [some error handling](https://github.com/mapbox/mapbox-studio/commit/f241d39a78c5ecbfc902d8de446223d8109e3617) for the mapnik call.

**Comment by springmeyer** [comment]
@GretaCB what value does this commit add? https://github.com/mapbox/mapbox-studio/commit/f241d39a78c5ecbfc902d8de446223d8109e3617. To me that just catches the Mapnik error and throws another. I'm biased, but personally I'd like to see the Mapnik error reported since it may contain helpful details to know what went wrong.

**Comment by GretaCB** [comment]
Good point. Will pass in the caught error.

**Comment by GretaCB** [comment]
@samanpwbb @springmeyer This is ready to merge. What is the process for deploy/release?

cc @rclark

**Comment by wilhelmberg** [comment]
My 2 cents concerning https://github.com/mapbox/mapbox-studio/commit/f241d39a78c5ecbfc902d8de446223d8109e3617

**If** I decide to catch an error and re-throw (happens rarely, most of the time to do some extra error logging), then I like to include the original error plus some additional information to provide more context and make debugging/error hunting easier, e.g.:

```
throw new Error(util.format(
    'Could not load datasource "%s" for layer "%s": %s'
    , opts
    , l.id
    , err
));
```

I'm not familiar with the style rules of this code base and haven't checked if there is a general `catch` somewhere, but maybe that's what @springmeyer meant, too.

**Comment by springmeyer** [comment]
> @samanpwbb @springmeyer This is ready to merge. What is the process for deploy/release?

Because this a fairly big upgrade I think it would be worth testing the published packages next. That looks like:

1) Publish packages by committing to this branch with `git commit -a -m "[publish mapnik-om-refactor]"

Details on getting the package urls at https://github.com/mapbox/mapbox-studio/blob/mb-pages/CONTRIBUTING.md#packaging.

2) Then test the OS X package locally (make sure it opens, can load some data), and then test the Windows package - plan over at https://github.com/mapbox/mapbox-studio/issues/1218.

@BergWerkGIS yes, exactly - that is what I meant - that at least keeping the original error is desirable, or adding to it like you've demonstrated.

**Comment by GretaCB** [comment]
So sorry @BergWerkGIS , I missed your response. Thanks for the example, I will commit a fix with the original error concatenated. 

Awesome plan, thanks @springmeyer !

File: app/source.js [hunk 1]
```diff
@@ -666,9 +663,10 @@ window.Source = function(templates, cwd, tm, source, revlayers, examples, isMapb
             success: function(metadata) {
                 // Clear loading state
                 $('#full').removeClass('loading');
-                var center = metadata.center;
-                var zoom = Math.max(metadata.minzoom, view.model.get('minzoom'));
-                map.setView([center[1], center[0]], zoom);
+                map.fitBounds([
```
Comment (samanpwbb) [inline app/source.js:666]: to account for https://github.com/mapbox/mapbox-studio/pull/1270 - fitBounds should disable animation as well.

**Comment by GretaCB** [comment]
#### Note to self:
- [x] disable animation in map.fitBounds...set `animation: false` (ref https://github.com/mapbox/mapbox-studio/pull/1270/files)

**Comment by springmeyer** [comment]
I'm stealing this as I realize I need this to be able to finish #1287. So, working on bringing this patch up to date now.

**Comment by springmeyer** [comment]
Successfully integrated this into #1287. Thanks @GretaCB!

**Pull request closed without merging**